feature/frontend_exam into develop

dhelmert opened the pull request:
### Reason for this PR
- Exams are now part of UC4, and need an api

### Changes in this PR
- Added exam-management.yaml
- Added examresult-management.yaml
- Updated admission-management to include exam admissions

dhelmert committed:
Add exam-management
Add examresult-management
Update admission-management to include exam admissions

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>
Co-authored-by: Robin Helmert <rhelmert@mail.uni-paderborn.de>

dhelmert committed:
Change query parameter to plura

bastihav commented on exam-management.yaml:
```diff
@@ -0,0 +1,310 @@
+              examples:
+                UnsignedProposalAddEntry:
+                  $ref: '#/components/examples/ExampleUnsignedProposal'
+          headers:
+            Location:
+              schema:
+                type: string
+                description: "URI of the newly created Admission"
```
```suggestion
```

bastihav commented on examresult-management.yaml:
```diff
@@ -0,0 +1,257 @@
+              examples:
+                UnsignedProposalAddEntry:
+                  $ref: '#/components/examples/ExampleUnsignedProposal'
+          headers:
+            Location:
+              schema:
+                type: string
+                description: "URI of the newly created Admission"
```
```suggestion
```

bastihav reviewed (changes requested):
- [x] for all lists parameters: specify that we need a comma seperated value
- [x] example description for UnsignedProposals are wrong (admission, exam, examresult)
- [x] specify "admittableAt" and "droppableAt" format -> ISO
- [x] specify admittableUntil and droppableUntil format in schema -> date iso

dhelmert committed:
Update exam-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update examresult-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Change all query parameter list descriptions to include that they are comma separated list
Change proposal example names and descriptions, to be more generic
Specify Date ISO whereever there are dates

dhelmert committed:
Add wrapper for post

NikoBergemann reviewed (commented):
Nothing out of the ordinary

dhelmert committed:
Multi calls in certificate

bastihav reviewed (changes requested):
we now have two superflous endpoints (for resolving single enrollmentId and single username respectively), which should be removed

dhelmert committed:
Remove superflous endpoints

dhelmert commented:
> 
> 
> we now have two superflous endpoints (for resolving single enrollmentId and single username respectively), which should be removed

Good point, the endpoints in certificate are now removed

dhelmert committed:
Rename ExamResult to be proper 2 words, camelCasing and snake_casing where appropiate

bastihav commented on exam-management.yaml:
```diff
@@ -0,0 +1,310 @@
+openapi: "3.0.0"
+info:
+  description: "This is the Exam API for UC4."
+  version: "0.16.1"
```
```suggestion
  version: "0.17.1"
```

bastihav commented on exam-result-management.yaml:
```diff
@@ -0,0 +1,266 @@
+openapi: "3.0.0"
+info:
+  description: "This is the Exam Result API for UC4."
+  version: "0.16.1"
```
```suggestion
  version: "0.17.1"
```

dhelmert committed:
Merge the AddAdmission endpoints into one
Add examTypes, grades to config

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

bastihav commented on admission-management.yaml:
```diff
@@ -1,7 +1,7 @@
 info:
   description: "This is the Admission API for UC4."
-  version: "0.15.1"
+  version: "0.16.1"
```
```suggestion
  version: "0.17.1"
```

dhelmert committed:
Update exam-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update exam-result-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Update admission-management.yaml

Co-authored-by: Bastian Haverkamp <bastihav@mail.upb.de>

dhelmert committed:
Merge branch 'develop' into feature/frontend_exam

# Conflicts:
#	admission-management.yaml
